lib/kamal/cli/app/boot.rb

```diff
@@ -64,4 +63,43 @@ def stop_old_version(version)
 
       execute *app.clean_up_assets if assets?
     end
+
+    def reach_barrier
+      if barrier
+        if barrier_role?
+          barrier.open
+        else
+          wait_for_barrier
+        end
+      end
```

Comment:
Early return suggestion:

```suggestion
      return unless barrier

      if barrier_role?
        barrier.open
      else
        wait_for_barrier
      end
```